documentation/projects/proposals/additional_search_views/20230719-implementation_plan_additional_search_views.md

```diff
@@ -0,0 +1,263 @@
+It should accept `href` as a prop.
+
+### Other interface changes
+
+#### Update the "by" area of the single result page
+
+The "by creator" line should be replaced with a section that is
+horizontally-scrollable on mobile, and contains the `VCreatorLink` and
+`VSourceLink` components.
+
+#### Update links in the "information" section
+
+Add external link icon to the provider link; add creator link with external link
+icon The links to creator in the image and audio single result pages should have
+an "external link" icon.
+[image creator link](https://github.com/WordPress/openverse/blob/35f08e5710d59f6c10f0cf54103fcce151adfefe/frontend/src/pages/image/_id/index.vue#L62-L73)
+and
+[audio creator link](https://github.com/WordPress/openverse/blob/35f08e5710d59f6c10f0cf54103fcce151adfefe/frontend/src/components/VAudioTrack/layouts/VFullLayout.vue#L32-L41).
+Audio creator link should also be updated to match the image creator link. It
+should be a conditional component: `VLink` if the `creator_url` is truthy and
+`span` if the `creator_url` is falsy.
+
+Currently, the `foreign_landing_url` is linked to the "source" in the image page
+and "provider" in the audio page. The audio page should be updated to match the
+image page: the `foreign_landing_url` link should be added to the "source", not
+provider.
+[Audio provider link](https://github.com/WordPress/openverse/blob/35f08e5710d59f6c10f0cf54103fcce151adfefe/frontend/src/components/VAudioDetails/VAudioDetails.vue#L61C1-L63),
+[image source link](https://github.com/WordPress/openverse/blob/35f08e5710d59f6c10f0cf54103fcce151adfefe/frontend/src/components/VImageDetails/VImageDetails.vue#L26-L28)
+
+### Store changes
+
+#### Update the `searchBy` filter in the `search` store
+
+Currently, this filter is shaped as other filters: it is a list of objects with
+`code`, `name` and `checked` properties. This is a possible value now: `{
+searchBy: { code: "creator", name: "filters.searchBy.creator", checked: true }}.
+
+Since this filter is mutually exclusive (you cannot search by both the creator
+and the tag, for example), this shape is very difficult to update.
+
+It is better to set the `searchBy` filter to have one of the possible values:
+`{ searchBy: <null|creator|source|tag> }`.
+
+We should also create a new method, `setSearchBy`, in the `search` store, that
+would allow directly set the `searchBy` filter value.
+
+### Analytics changes
+
+I am not sure what events we want to add, if any. Should we add `<VIEW>_CLICKED`
+events for the new views, or would page views be sufficient?
+
+Should we add a new `SELECT_COLLECTION_ITEM` event similar to
+`SELECT_SEARCH_RESULT`?
+
+The clicks on external creator link in the `VCollectionHeader` should be tracked
+as `VISIT_CREATOR_LINK` events. We don't have a special event for visiting
+source, should we add one, or should they be tracked as generic
+`VISIT_EXTERNAL_LINK`?
```

Comment:
It makes sense to me to add an event for these because we might want to know how these links are used broadly ("how many provider link clicks are there?") rather than specifically ("how many clicks to https://flickr.com are there?").